### PR TITLE
Framework: Switch menus tests to single runner

### PR DESCRIPTION
**Pull request by seear**
This went mostly smoothly. Had to introduce the use of filesystem mocks into the `item-options` test to avoid `double callback!` hangs.

### CODE REVIEW

File: client/my-sites/menus/item-options/test/test-posts.jsx [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
+import { useSandbox } from 'test/helpers/use-sinon';
+import useFilesystemMocks from 'test/helpers/use-filesystem-mocks';
+
+describe( 'Posts', function() {
+	let Posts, siteMenus, helpers;
+	useFakeDom();
+	useSandbox();
+
+	const menuDataPath = path.dirname( require.resolve( 'lib/menu-data' ) );
+	useFilesystemMocks( menuDataPath + '/test' );
```
Comment (seear) [inline client/my-sites/menus/item-options/test/test-posts.jsx]: Technique here for picking up filesystem mocks from elsewhere is convoluted, so maybe we can add something into the helper for this if it becomes common.

Comment (blowery) [inline client/my-sites/menus/item-options/test/test-posts.jsx]: i think this is actually OK. I like that it requires a bit of gymnastics to do something odd.

**Comment by blowery** [comment]
looks good! :ship:

**Comment by gziolo** [comment]
I added one tiny change. We started to name all test files after file under test. We already have `test` folder, and we want to keep only test files directly in test folder :)
Thanks for porting more tests :+1:

**Comment by blowery** [comment]
Thanks @gziolo I missed that.  Good catch!

**Pull request merged**